spock-core/src/main/java/org/spockframework/compiler/AbstractDeepBlockRewriter.java

```diff
@@ -30,9 +30,10 @@
   protected MethodCallExpression currMethodCallExpr;
   protected ClosureExpression currClosure;
   protected ISpecialMethodCall currSpecialMethodCall = NoSpecialMethodCall.INSTANCE;
-  protected Statement lastSpecialMethodCallStat;
+  protected Collection<Statement> pastSpecialMethodCallStats = new ArrayList<>();
```

Comment:
If we only use this for contains checks would a set make more sense?

Reply:
How you like. I usually only use a `Set` if I want to ensure that I only have each entry once. But I don't care if it is contained multiple times, I just care of whether it is contained at all and thus was already handled. Apart from that, it shouldn't actually be possible to happen that a statement is in there mutliple times I think.

Reply:
When I only need to check if something has been already seen I normally prefer a `Set` as this uses hash based lookup instead of linear search, in this case it might not matter much as the data structure is a bit more expensive.

Reply:
Ah, you mean `HashSet` instead of `ArrayList`, not `Set` instead of `Collection`.
Yeah, you are probably right about the hash-based lookup, changed it.